src/tests/ftest/cart/test_group_np_common.h

```diff
@@ -146,48 +149,43 @@ test_swim_status_handler(crt_rpc_t *rpc_req)
 	regex_t				 regex_dead;
 	static const char		*dead_regex = ".?0*1";
 	static const char		*alive_regex = ".?0*";
-	char				swim_seq[MAX_SWIM_STATUSES];
-	size_t				sc = sizeof(char);
+	int				rank;
 
 	/* CaRT internally already allocated the input/output buffer */
 	e_req = crt_req_get(rpc_req);
-
-	if (swim_status_by_rank[e_req->rank] != NULL)
-		memcpy(swim_seq,
-		       swim_status_by_rank[e_req->rank],
-		       strlen(swim_status_by_rank[e_req->rank]) * sc);
-	else
-		memset(swim_seq, 0x0, MAX_SWIM_STATUSES);
+	rank = e_req->rank;
 
 	/* compile and run regex's */
 	regcomp(&regex_dead, dead_regex, REG_EXTENDED);
 	int rc_dead = regexec(&regex_dead,
-			      swim_seq,
+			      swim_seq_by_rank[rank],
 			      0, NULL, 0);
 	regcomp(&regex_alive, alive_regex, REG_EXTENDED);
 	int rc_alive = regexec(&regex_alive,
-			       swim_seq,
+			       swim_seq_by_rank[rank],
 			       0, NULL, 0);
 
 	D_ASSERTF(e_req != NULL, "crt_req_get() failed. e_req: %p\n", e_req);
```

Comment:
Move up  to where e_req is first found.   Why print it out on failure if it fails only for 0.

Reply:
f85bdd3c5a25acbc7c69af747ddd54fa72438148